### DISCUSSION TITLE
Is homespan breaking esp-now communication?

### DISCUSSION THREAD

**Discussion by zhekch** [HomeSpan/HomeSpan#420] [Q&A]
I'm trying to implement esp communication and saw this, [esp-now two way communication](https://randomnerdtutorials.com/esp-now-two-way-communication-esp32/), but apparently homespan is not capable with it, because adding `homeSpan.poll();` makes the code not work. Are there any other ways I can send data from one esp to another without the need of making each esp as a separate accessory? I wanted to make one esp32 act as a server, and other esp8266 sending data to server

**Comment by blaisebarrette** [comment]
ESP now support is currently being added to Homespan and will be available in the next release. :)
Look for NOW.md in the docs folder of the dev branch to read how it’s gonna be implemented.

**Comment by HomeSpan** [comment]
That's correct - HomeSpan now includes a dedicated class, *SpanPoint*,  that implements the ESP-NOW protocol.  Note it is written for the ESP32 and unfortunately will not work on an ESP8266.  This new class is currently in the **dev** branch and will be part of the next release.  Feel free to try it out now in dev.

The reason why you found your code did not work once `homeSpan.poll()` was called is that ESP-NOW uses the same WiFi radio that is used to connect the ESP32 to your home network.  This connection is made when `homeSpan.poll()` is first called.   It is possible to use both ESP-NOW and connect to a WiFi network but extra logic is needed to ensure ESP-NOW uses the same WiFi channel that the ESP32 is using to connect to your home network.  There are also some other configuration parameters that need to be properly set.  The new *SpanPoint* class does all of this for you automatically.

**Reply by zhekch** [reply]
So are there any known ways to send sensor data from esp8266 to esp32, and then to HomeKit?

**Reply by homonto** [reply]
I know 8266 is not the same as C3, but I can confirm as I tested that these MCU work: ESP32, ESP32-S, S2, C3, S3
I did not test 8266 as I don't have one.

**Reply by HomeSpan** [reply]
I have just added new functionality that allows HomeSpan to receive ESP-NOW data from an ESP8266.  The new functionality (which consists of a 5th optional parameter in the SpanPoint constructor) is in the dev branch --- see the dev branch version of the SpanPoint page for details.  I also included on that page a detailed example sketch, *RemoteDevice8266*, showing how to implement ESP-NOW on an ESP8266 so it can properly communicate with an ESP32 running HomeSpan.

**Reply by homonto** [reply]
ha, so all "esphome" sockets and bulbs, with esp8266 inside can work with HomeSpan?
that would be a breakthrough ;-)
